### PR TITLE
CDAP-6743 Check that the custom mapped namespace directory is empty

### PR DESCRIPTION
**Pull request by rohitsinha54**
Issue: https://issues.cask.co/browse/CDAP-6743
Build: http://builds.cask.co/browse/CDAP-DUT4533-3

### CODE REVIEW

**Comment by anwar6953** [comment]
If this is something we enforce, don't we practically own the directory? Why don't we perform creation/deletion of that directory upon ns create/delete?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java [hunk 1]
```diff
@@ -140,6 +140,14 @@ private void createLocation(NamespaceMeta namespaceMeta) throws IOException {
             "with sufficient privileges for the user and then try creating a namespace.",
           customNamespacedLocation.toString(), namespaceMeta.getNamespaceId()));
       }
+      // we also expect it to empty since non-empty directories can lead to various inconsistencies CDAP-6743
+      if (customLocation.list().length != 0) {
+        // TODO: Add username in the below exception message.
+        throw new IOException(String.format(
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java:145]: why do we need username?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java [hunk 2]
```diff
@@ -140,6 +140,14 @@ private void createLocation(NamespaceMeta namespaceMeta) throws IOException {
             "with sufficient privileges for the user and then try creating a namespace.",
           customNamespacedLocation.toString(), namespaceMeta.getNamespaceId()));
       }
+      // we also expect it to empty since non-empty directories can lead to various inconsistencies CDAP-6743
+      if (customLocation.list().length != 0) {
+        // TODO: Add username in the below exception message.
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java]: why is this a TODO still? we already have this info in the `NamespaceMeta` right?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java [hunk 3]
```diff
@@ -140,6 +140,14 @@ private void createLocation(NamespaceMeta namespaceMeta) throws IOException {
...
+        throw new IOException(String.format(
+          "The provided home directory '%s' for namespace '%s' is not empty. Namespace can only be mapped to an " +
+            "empty directory. Please try creating the namespace again with an empty directory mapping.",
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java]: nit: the second sentence may not add much value. but feel free to ignore my comment.

**Comment by bdmogal** [comment]
couple comments. rest LGTM

**Comment by rohitsinha54** [comment]
@anwar6953 : No, we don't own the directory. Creation is managed by the administrator of the cluster and then permissions is given to appropriate user to read/write to it for a project or something similar.

**Comment by rohitsinha54** [comment]
Build passed. Merging.

**Pull request merged**